substrate/client/authority-discovery/src/worker/schema/dht-v3.proto

```diff
@@ -0,0 +1,33 @@
+syntax = "proto3";
+
+package authority_discovery_v3;
+
+// First we need to serialize the addresses in order to be able to sign them.
+message AuthorityRecord {
+	// Possibly multiple `MultiAddress`es through which the node can be 
+	repeated bytes addresses = 1;
+}
+
+message PeerSignature {
+	bytes signature = 1;
+	bytes public_key = 2;
+}
+
+// Information regarding the creation data of the record
+message TimestampInfo {
+	// Time since UNIX_EPOCH in nanoseconds, scale encoded
+	bytes timestamp = 1;
+	// A signature of the creation time.
+	bytes signature = 2;
+}
+
+// Then we need to serialize the authority record and signature to send them over the wire.
+message SignedAuthorityRecord {
+	bytes record = 1;
+	bytes auth_signature = 2;
+	// Even if there are multiple `record.addresses`, all of them have the same peer id.
+    // Old versions are missing this field. It is optional in order to provide compatibility both ways.
+	PeerSignature peer_signature = 3;
```

Comment:
I think we can make the peer signature adn timestamp non-optional and also close: https://github.com/paritytech/polkadot-sdk/issues/3824.

Reply:
I would keep that separate.